### PR TITLE
Support for macro params in filter operands

### PR DESCRIPTION
**Pull request by saqimtiaz**
@Jermolene this adds support for macro parameters in filters as discussed in #5830. 

Salient changes:
- refactored `$tw.utils.parseMacroInvocation()` to create `$tw.utils.parseMacroParameters()` which can be re-used for parsing macro parameters.
  - We have existing tests for parsing macros that show this is change is backwards compatible. 
- Introduced `$tw.utils.parseFilterVariable()` for which I have added tests.

### CODE REVIEW

File: core/modules/parsers/parseutils.js [hunk 1]
```diff
@@ -208,6 +215,29 @@ exports.parseMacroInvocation = function(source,pos) {
 	return node;
 };
 
+exports.parseFilterVariable = function(source) {
+	var node = {
+			name: "",
+			params: [],
+		},
+		pos = 0,
+		reName = /([^\s>"'=]+)/g;
```
Comment (Jermolene) [inline core/modules/parsers/parseutils.js:224]: Why do we use `=` as a terminator?

**Review by Jermolene — CHANGES_REQUESTED** [review]
Thanks @saqimtiaz that was quick! Just a question, but also there's a bunch of whitespace changes which it would be great to remove for readability.

File: core/modules/parsers/parseutils.js (hunk 1 above)
Comment (saqimtiaz) [reply core/modules/parsers/parseutils.js:224]: @Jermolene I re-used the regular expression from [parseMacroInvocation](https://github.com/Jermolene/TiddlyWiki5/blob/master/core/modules/parsers/parseutils.js#L174) and I spotted this but left it alone as I wasn't completely sure why we had it in that context either. Will revise.

**Comment by saqimtiaz** [comment]
Apologies for the spurious white space changes, will resolve. Working on a different machine today and thus a new text editor, which has a mind of its own.

**Comment by saqimtiaz** [comment]
- regular expression fixed.
- added more tests for edge cases.
- white space changes have been resolved (I really wonder why an editor would default to making whitespace changes unprompted!)

**Comment by saqimtiaz** [comment]
Note that the following is also possible `[<now [UTC]YYYY0MM0DD0hh0mm0ssXXX>] `.

I've added a few tests specifically for square brackets in macro params.

**Comment by pmario** [comment]
So this PR enables every macro to be used with params. right? Since it uses the macro parameter parser it's also possible to use named params?  eg: `[<now format:"[UTC]YYYY0MM0DD0hh0mm0ssXXX">] ` Is that possible?

**Comment by saqimtiaz** [comment]
> So this PR enables every macro to be used with params. right? Since it uses the macro parameter parser it's also possible to use named params?  eg: `[<now format:"[UTC]YYYY0MM0DD0hh0mm0ssXXX">] ` Is that possible?

Yes. See tests.

**Comment by Jermolene** [comment]
Thanks @saqimtiaz I think this might open up some interesting capabilities. It certainly makes it easier to bring custom JS code into filters.
